test/systemtest/functional_fpga_test.py

```diff
@@ -53,13 +53,22 @@ def spiflash_proc(self, tmp_path, spiflash, sw_test_bin):
 
         p_flash.terminate()
 
-    def test_execute_binary(self, spiflash_proc, fpga_uart, uart_timeout):
+    def test_execute_binary(self, spiflash_proc, fpga_uart, uart_timeout, logfile):
         """
         Executes the binary and inspects its UART for "PASS!\r\n" or "FAIL!\r\n".
         """
 
         logger = logging.getLogger(__name__)
 
+        if logfile:
+            logger.debug("Logfile at" + logfile)
+            # create logger with 'spam_application'
+            logger.setLevel(logging.DEBUG)
+            # create file handler which logs even debug messages
+            fh = logging.FileHandler(filename=logfile, mode='w')
+            fh.setLevel(logging.DEBUG)
+            logger.addHandler(fh)
```

Comment:
This block is repeated twice here and in the Verilator fixture. Consider factoring this into a helper function in test_util?

Reply:
moved